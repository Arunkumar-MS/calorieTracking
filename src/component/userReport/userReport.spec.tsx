import { jest } from '@jest/globals'
import { shallowToJson } from "enzyme-to-json";
import * as React from 'react';
import { shallow } from 'enzyme';
import UserReport from './index';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: () => mockDispatch
}));


jest.mock("swr", () => ({
    __esModule: true,
    namedExport: jest.fn(),
    default: () => ({
        data: {"02:02:2022":685.0600000000001,"01:02:2022":1196.15,"03:02:2022":3262.5,"29:01:2022":303.4,"30:01:2022":189,"31:01:2022":201.6},
    }),
}));

describe('UserReport page', () => {
    it('should render properly', function () {
        const wrap = shallow(<UserReport />)
        expect(shallowToJson(wrap)).toMatchSnapshot();
    })
})
