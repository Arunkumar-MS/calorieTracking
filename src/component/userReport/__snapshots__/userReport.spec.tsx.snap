// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserReport page should render properly 1`] = `
<div>
  <div
    data-test-id="report-page-title"
  >
     Your last 7 days calorie consumption report  
  </div>
  <ForwardRef(LoadableComponent)
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": Array [
              "green",
              "green",
              "green",
              "green",
              "green",
              "green",
            ],
            "data": Array [
              201.6,
              189,
              303.4,
              3262.5,
              1196.15,
              685.0600000000001,
            ],
          },
        ],
        "labels": Array [
          "31:01:2022",
          "30:01:2022",
          "29:01:2022",
          "03:02:2022",
          "01:02:2022",
          "02:02:2022",
        ],
      }
    }
    options={
      Object {
        "plugins": Object {
          "legend": null,
          "title": Object {
            "display": true,
            "text": "Last 7 days report",
          },
        },
        "responsive": true,
      }
    }
  />
</div>
`;
